### ISSUE TITLE
ak2i troubles

### ISSUE THREAD

**Issue by zoogie** [ntrteam/dsfirmverify#1]
I have 2 ak2i's (HW 81) flashed with ntrboot and verified manually that they work. When running them with dsfirmveryify, however, the result is the same;

ncgc_nbegin_key1 failed:
-2 0x00000FC2 0x7EAA0570

**Comment by angelsl** [comment]
Yeah, I am aware of this &mdash; the AK2i does some kind of detection to determine whether to serve the NTR part of its flash or the TWL part (FIRM is written to the TWL part) but this currently gets the NTR part of the flash, which contains the normal Acekard 2i title (and normal NTR key).

I'll work on it.

**Comment by pixel-stuck** [comment]
@angelsl likely it's just the init order. Iirc, two commands in the initial set of them are actually flipped on TWL vs NTR

Yep:
DS-Lite cart initialization sequence:
CMD 9f00000000000000 # reset
CMD 0000000000000000 # get header
CMD 9000000000000000 # get rom_id
CMD ...

DSi cart init sequence:
CMD 9f00000000000000 # reset
CMD 9000000000000000 # get rom_id
CMD 0000000000000000 # get header
CMD ...

Source: https://hackmii.com/2010/02/lawsuit-coming-in-3-2-1/

EDIT: looks like you handle this already too with ncgc_ninit_order()

**Comment by angelsl** [comment]
@dark-samus 

Tried that, but no difference (?)

The funny thing is, Boot9 actually inits like a DS phat/lite.
